app/controllers/datasets_controller.rb

```diff
@@ -4,9 +4,16 @@ class DatasetsController < ApplicationController
   include DatasetsHelper
 
   def show
+    begin
+      @dataset = current_dataset
+      raise 'Metadata missing' if @dataset['title'].empty?
+
+    rescue
+      render :template => "errors/not_found", :status => 404
+    end
```

Comment:
It would be nice to have a different 404 body message, to differentiate real 404s from metadata-missing 404s. Or we can find another http response code that matches the semantics.